### PR TITLE
[kubetest] Initial DumpClusterLogs for Azure

### PR DESCRIPTION
**Pull request by papagalu**

### CODE REVIEW

**Comment by test-e2e-win** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approvers: 

If they are not already assigned, you can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/e2e-win/test-infra/blob/azure_provider/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by test-e2e-win** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: 
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approvers: 

Assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/e2e-win/test-infra/blob/azure_provider/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by test-e2e-win** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approvers: 

If they are not already assigned, you can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/e2e-win/test-infra/blob/azure_provider/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by test-e2e-win** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: 
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approvers: 

Assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/e2e-win/test-infra/blob/azure_provider/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: kubetest/azure.go [hunk 1]
```diff
@@ -537,7 +537,36 @@ func (c Cluster) Down() error {
...
+	// should run on master node and collect logs through winrm from windows slaves
+
+	const AzureLogDumpTemplate = `
+function wrapper-ssh () {
+	ssh -i %[1]s -o StrictHostKeyChecking=no "$@"
+}
+function wrapper-scp () {
+	scp -i %[1]s -o StrictHostKeyChecking=no "$@"
+}
+wget https://raw.githubusercontent.com/e2e-win/e2e-win-prow-deployment/master/custom-scripts/master.sh
```
Comment (adelina-t) [inline kubetest/azure.go]: I wouldn't have this dependency here. Why not just use the masterscript as a template too ?

File: kubetest/azure.go [hunk 2]
```diff
@@ -537,7 +537,36 @@ func (c Cluster) Down() error {
...
+wrapper-ssh azureuser@%[2]s 'cat | bash /dev/stdin --user azureuser --pass Passw0rdAdmin --output /tmp/out' < master.sh
```
Comment (adelina-t) [inline kubetest/azure.go]: Do not use user and pass hardcoded.

File: kubetest/azure.go [hunk 3]
```diff
@@ -537,7 +537,36 @@ func (c Cluster) Down() error {
...
+wrapper-scp -r azureuser@%[2]s:/tmp/out/ %[3]s
+`
+	// Prevent an obvious injection.
+	if strings.Contains(localPath, "'") || strings.Contains(gcsPath, "'") {
+		return fmt.Errorf("%q or %q contain single quotes - nice try", localPath, gcsPath)
```
Comment (adelina-t) [inline kubetest/azure.go]: Copy pasta with comments. Nice :)

File: kubetest/azure.go [hunk 4]
```diff
@@ -537,7 +537,36 @@ func (c Cluster) Down() error {
...
+	}
+
+	var masterHostname string
+	masterHostname = fmt.Sprintf("%s.%s.cloudapp.azure.com", c.dnsPrefix, c.location)
```
Comment (adelina-t) [inline kubetest/azure.go]: Extract dns name template into a variable. Much cleaner when we need to change it for some reason.

**Comment by adelina-t** [comment]
One other things that came to mind now. Try to add the kubernetes.json template to the logs as well, trim sensitive parts. Should be easy enough to do in go.

File: kubetest/azure.go (hunk 3 above)
Comment (adelina-t) [reply kubetest/azure.go]: Also. who is gcsPath ?

**Comment by test-e2e-win** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approvers: 

If they are not already assigned, you can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/e2e-win/test-infra/blob/azure_provider/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by test-e2e-win** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approvers: 

If they are not already assigned, you can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/e2e-win/test-infra/blob/azure_provider/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by test-e2e-win** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: 
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approvers: 

Assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/e2e-win/test-infra/blob/azure_provider/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by test-e2e-win** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To fully approve this pull request, please assign additional approvers.
We suggest the following additional approvers: 

If they are not already assigned, you can assign the PR to them by writing `/assign ` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/e2e-win/test-infra/blob/azure_provider/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**